core/src/main/scala/org/apache/spark/rdd/ShuffledRDD.scala

```diff
@@ -104,10 +105,26 @@ class ShuffledRDD[K: ClassTag, V: ClassTag, C: ClassTag](
   }
 
   override def compute(split: Partition, context: TaskContext): Iterator[(K, C)] = {
+    // Use -1 for our Shuffle ID since we are on the read side of the shuffle.
+    val shuffleWriteId = -1
+    // If our task has data property accumulators we need to keep track of which partitions
+    // we are processing.
+    if (context.taskMetrics.hasDataPropertyAccumulators()) {
+      context.setRDDPartitionInfo(id, shuffleWriteId, split.index)
+    }
     val dep = dependencies.head.asInstanceOf[ShuffleDependency[K, V, C]]
-    SparkEnv.get.shuffleManager.getReader(dep.shuffleHandle, split.index, split.index + 1, context)
+    val itr = SparkEnv.get.shuffleManager.getReader(dep.shuffleHandle, split.index, split.index + 1,
+      context)
```

Comment:
don't you need to wrap this iterator to reset the partitionInfo per record, like you do in `MapPartitionsRDD`?  what if there is `rdd.reduceByKey(...).map(...)` with accumulator updates in both?

The more I think about this, I am wondering if dataproperty accumulators should only be supported in map partitions rdd.  (sorry I think long ago I argued for putting this in ... I am still not certain ...)